### PR TITLE
linux: further cleanup config after drop of 4.9

### PR DESCRIPTION
**Pull request by fabianhjr**
###### Description of changes

Continuation of #201845

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/kernel/manual-config.nix [hunk 1]
```diff
@@ -55,8 +55,7 @@ let
     hasAttr getAttr optional optionals optionalString optionalAttrs maintainers platforms;
 
   # Dependencies that are required to build kernel modules
-  moduleBuildDependencies = [ perl ]
-    ++ optional (lib.versionAtLeast version "4.14") libelf
+  moduleBuildDependencies = [ libelf perl ]
```
[the remark this answers is not in the document]
Comment (fabianhjr) [reply pkgs/os-specific/linux/kernel/manual-config.nix:58]: Oh right, yeah I was wondering were the rebuilds were coming from.

Amending

[the remark this answers is not in the document]
Comment (fabianhjr) [reply pkgs/os-specific/linux/kernel/manual-config.nix]: Yep, that was it: 0 rebuilds now.

I conceptualized lists as being treated like sets for rebuild purposes, was a bit wrong there. :sweat_smile:

Thanks for the nit/pointer. :3

**Review by vcunat — APPROVED** [review]

**Review by trofi — APPROVED** [review]

**Comment by fabianhjr** [comment]
Hi, could this merge request get merged @vcunat ?

**Comment by vcunat** [comment]
Right, I think we've given enough time for further feedback.

**Comment by fabianhjr** [comment]
Ty :3

**Pull request merged**